packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/Providers/MethodProvider.cs

```diff
@@ -100,6 +102,11 @@ public void Update(
             }
 
             Signature = updated.Signature;
+            var updatedParameterNames = Signature.Parameters.Select(p => p.Name).ToList();
+            if (!originalPartaMeterNames.SequenceEqual(updatedParameterNames) || XmlDocs is null)
```

Comment:
adding XmlDocs.IsEmpty into the condition will introduce regen diff

Reply:
There might be other case that we should rebuild xmlDocs, which can be added later. My primary goal for this PR is to unblock https://github.com/Azure/azure-sdk-for-net/pull/50776.

Reply:
If we want to reuse XmlDoc.
`if (XmlDoc is null || XmlDoc == XmlDocHelper.InheritDocs || XmlDoc == XmlDoc.Empty` should work.

Reply:
Should be `!Signature.Equals(updated.Signature)`

Reply:
Also, I think we would need to pass the `MethodSignatureComparer`

Reply:
Since it is always the same instance, which is I think a good thing, we should remove the signature assignment in L104.